packages/back-end/src/queryRunners/QueryRunner.ts

```diff
@@ -191,13 +213,77 @@ export abstract class QueryRunner<
     });
   }
 
+  public async startReadyQueries(): Promise<void> {
+    const queuedQueries = await getQueriesByIds(
+      this.model.organization,
+      this.model.queries
+        .filter((q) => q.status === "queued")
+        .map((q) => q.query)
+    );
+    logger.debug(
+      `Starting any queued queries that are ready: ${queuedQueries.map(
+        (q) => q.id
+      )}`
+    );
+    queuedQueries.map(async (query) => {
```

Comment:
We should await this, otherwise any errors throw inside will not be logged properly.

```ts
await Promise.all(queuedQueries.map(async (query) => {
```